### PR TITLE
DF-3266 BMC: update requirements and troubleshooting

### PR DESCRIPTION
**Pull request by mhofert-r7**
### Description

Describe the proposed changes:

  - update help.md to improve requirements and troubleshooting

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by mhofert-r7** [comment]
Please note that Service related actions were never implemented. As such including documentation about them seems nonsensical even harmful IMO

File: bmc_remedy_itsm/help.md [hunk 1]
```diff
@@ -1304,6 +1306,7 @@ _This plugin does not contain any custom output types._
 
 By default the API port is 8008.
 To configure the BMC server for API usage: https://docs.bmc.com/docs/ars9000/configuring-the-rest-api-529403222.html
+All testing was done on BMC Remedy ITSM 9.1.5 With MS SQL 2016 as the database.
```
Comment (gusingh-r7) [inline bmc_remedy_itsm/help.md]:
```suggestion
All testing was done on BMC Remedy ITSM 9.1.5 with MS SQL 2016 as the database.
```

Comment (jschipp-r7) [reply bmc_remedy_itsm/help.md]: Good find!

File: bmc_remedy_itsm/help.md [hunk 2]
```diff
@@ -1304,6 +1306,7 @@ _This plugin does not contain any custom output types._
 
 By default the API port is 8008.
 To configure the BMC server for API usage: https://docs.bmc.com/docs/ars9000/configuring-the-rest-api-529403222.html
+All testing was done on BMC Remedy ITSM 9.1.5 with MS SQL 2016 as the database.
```
Comment (zyoutz-r7) [inline bmc_remedy_itsm/help.md]: Make these bullets to easier visualize?

File: bmc_remedy_itsm/help.md [hunk 3]
```diff
@@ -1304,6 +1306,7 @@ _This plugin does not contain any custom output types._
 
 By default the API port is 8008.
 To configure the BMC server for API usage: https://docs.bmc.com/docs/ars9000/configuring-the-rest-api-529403222.html
```
Comment (zyoutz-r7) [inline bmc_remedy_itsm/help.md]: We might want to call out specifically the use of the REST api as many are still using the SOAP api.

Comment (mhofert-r7) [reply bmc_remedy_itsm/help.md]: damn good call I completely forgot about the 2 different apis

**Pull request merged**